skpro/regression/linear/_glm.py

```diff
@@ -227,31 +440,53 @@ def _fit(self, X, y):
         """
         from statsmodels.genmod.generalized_linear_model import GLM
 
+        # remove the offset and exposure columns which
+        # was inserted to maintain the shape
+        offset_var = self._offset_var
+        exposure_var = self._exposure_var
+
+        if offset_var is not None:
+            if isinstance(offset_var, int):
+                offset_var = pd.Index([X.iloc[:, offset_var].name])
+        if exposure_var is not None:
+            if isinstance(exposure_var, int):
+                exposure_var = pd.Index([X.iloc[:, exposure_var].name])
+        if offset_var is not None and exposure_var is not None:
+            X = X.drop([offset_var[0], exposure_var[0]], axis=1)
+        elif offset_var is not None:
+            X = X.drop(offset_var, axis=1)
+        elif exposure_var is not None:
+            X = X.drop(exposure_var, axis=1)
+
         X_ = self._prep_x(X)
 
         y_col = y.columns
 
+        family = self._family
+        link = self._link
+        sm_family = self._str_to_sm_family(family=family, link=link)
+
         glm_estimator = GLM(
             endog=y,
             exog=X_,
-            family=self._family,
-            missing=self.missing,
+            family=sm_family,
+            missing=self._missing,
         )
 
         self._estimator = glm_estimator
 
         fitted_glm_model = glm_estimator.fit(
-            self.start_params,
-            self.maxiter,
-            self.method,
-            self.tol,
-            self.scale,
-            self.cov_type,
-            self.cov_kwds,
-            self.use_t,
-            self.full_output,
-            self.disp,
-            self.max_start_irls,
+            self._start_params,
```

Comment:
I would say, prepare a dictionary for this and use kwargs, not args. Using so many args is quite risky.